packages/navigator/src/electron-browser/electron-navigator-menu-contribution.ts

```diff
@@ -0,0 +1,77 @@
+/********************************************************************************
+ * Copyright (C) 2021 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Command, CommandContribution, CommandRegistry, MenuContribution, MenuModelRegistry } from '@theia/core';
+import { CommonCommands, KeybindingContribution, KeybindingRegistry } from '@theia/core/lib/browser';
+import { WidgetManager } from '@theia/core/lib/browser/widget-manager';
+import * as electron from '@theia/core/shared/electron';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { FileStatNode } from '@theia/filesystem/lib/browser';
+import { FileNavigatorWidget, FILE_NAVIGATOR_ID } from '../browser';
+import { NavigatorContextMenu } from '../browser/navigator-contribution';
+import { isWindows, isOSX } from '@theia/core/lib/common/os';
+
+export const OPEN_CONTAINING_FOLDER = Command.toDefaultLocalizedCommand({
+    id: 'revealFileInOS',
+    category: CommonCommands.FILE_CATEGORY,
+    label: isWindows ? 'Reveal in File Explorer' :
+        isOSX ? 'Reveal in Finder' :
+        /* linux */ 'Open Containing Folder'
+});
+
+@injectable()
+export class ElectronNavigatorMenuContribution implements MenuContribution, CommandContribution, KeybindingContribution {
+
+    @inject(WidgetManager)
+    protected readonly widgetManager: WidgetManager;
+
+    registerCommands(commands: CommandRegistry): void {
+        commands.registerCommand(OPEN_CONTAINING_FOLDER, {
+            isEnabled: () => this.getSelectedFileStatNodes().length > 0,
+            isVisible: () => this.getSelectedFileStatNodes().length > 0,
+            execute: () => {
+                this.getSelectedFileStatNodes().forEach(node => {
+                    electron.shell.showItemInFolder(node.uri['codeUri'].fsPath);
```

Comment:
I actually just noticed another issue on Windows. Although it does work (opening the explorer with the file selected), Windows does not focus the explorer and opens it as a background window. This seems to be a known issue (see https://github.com/electron/electron/issues/4349).

Perhaps this might be fixed when upgrading to a newer version of Electron, however the workaround is to simply call `electron.remote.shell` instead of `electron.shell`. I was able to confirm locally that this fixes the issue. If this does not introduce new issues on Ubuntu/Mac, do you mind adding this fix?

Reply:
Thanks! I updated the change, the workaround does not seem to have any side effects in Ubuntu.